new mantle convection cookbook (#3798)
geodynamics/aspect: doc/manual/manual.tex

We can observe that when the two-layer viscosity model is added, there is only one apparent peak for each case. The first peaks of the 0.01 viscosity contrast tests appear earlier and are larger in magnitude than those of 0.1 viscosity contrast tests.  There are no secondary plumes and the whole system tends to reach stability after around 500 time steps.

> \subsubsection{The Bunge et al. mantle convection experiments}

Reviewer: ```suggestion
\subsubsection{The Bunge et al.~mantle convection experiments}
```

\textit{This section was contributed by Cedric Thieulot and Bob Myhill.}

> Early mantle modelling studies of the 70's, 80's and 90's were often concerned with simple setups (Cartesian geometries, incompressible fluid, free slip boundary conditions) and investigated the influence of the Rayleigh number, the heating mode or the temperature dependence of the viscosity on temperature, pressure and/or strain rate \cite{youn74,buss75,buss79,BBC89,BC93,vavy93,burb97}. In this cookbook, we use the `simple' material model to reproduce the setup in \cite{burb96}, which reported that even modest increases in mantle viscosity with depth could have a marked effect on the style of mantle convection.

Reviewer: ```suggestion
Early mantle modeling studies of the 1970s, 1980s and 1990s were often concerned with simple set-ups (Cartesian geometries, incompressible fluid, free slip boundary conditions) and investigated the influence of the Rayleigh number, the heating mode, or the temperature dependence of the viscosity on temperature, pressure and/or strain rate \cite{youn74,buss75,buss79,BBC89,BC93,vavy93,burb97}. In this cookbook, we use the `simple' material model to reproduce the setup in \cite{burb96}, which reported that even modest increases in mantle viscosity with depth could have a marked effect on the style of mantle convection.
```

Early mantle modelling studies of the 1970s, 1980s and 1990s were often concerned with simple set-ups (Cartesian geometries, incompressible fluid, free slip boundary conditions) and investigated the influence of the Rayleigh number, the heating mode or the temperature dependence of the viscosity on temperature, pressure and/or strain rate \cite{youn74,buss75,buss79,BBC89,BC93,vavy93,burb97}. In this cookbook, we use the `simple' material model to reproduce the set-up in \cite{burb96}, which reported that even modest increases in mantle viscosity with depth could have a marked effect on the style of mantle convection.

> 

Reviewer 2: Would you mind adding a sentence that the benchmark is ``\url{cookbooks/bunge_et_al_mantle_convection/bunge_et_al.prm}``?

> Although the original article showcases results obtained in a 3D hollow sphere, we here run the models in an annular domain of inner radius R$_i$=3480 \si{\kilo\meter} and outer radius  R$_o$ = 6370 \si{\kilo\meter}.

Reviewer: ```suggestion
Although the original article showcases results obtained in a 3D hollow sphere, we here run the models in an annular domain of inner radius $R_i$=3480 \si{\kilo\meter} and outer radius  $R_o$ = 6370 \si{\kilo\meter}.
```

> Although the original article showcases results obtained in a 3D hollow sphere, we here run the models in an annular domain of inner radius $R_i = 3480$ \si{\kilo\meter} and outer radius  $R_o = 6370$ \si{\kilo\meter}.

Reviewer 2: Here and in many places below: I am a bit confused, in the rest of the manual we have the ``\si{...}`` inside the math environments. Is there a reason you are not doing this? Does this look the same?

Reviewer 3: The units themselves look identical inside or outside the math environment (because \si overrides the local environment).  The spacing between number and \si{unit} is not necessarily the same, since the math environment typically imposes narrower whitespace. I prefer having \si outside the math environment, which consistently imposes a normal-width space between number and unit. 

The functions provided by siunitx always work fine outside the math environment, and mostly work inside the math environment. There are a few functions that only work outside the math environment. For example, the \numlist "function should be used in text mode, a common feature of all of the list and range functions provided by siunitx.". 

As an aside, there are some pretty odd uses of \si in the ASPECT manual right now ... like this one:
$\frac{\textrm{J}}{\si{kg}\cdot\si{K}} = \frac{\si{m}^2}{\textrm{s}^2\cdot\si{K}}$.

which I would write as \si{\joule\per\kilo\gram\per\kelvin} = \si{\meter\squared\per\second\squared\per\kelvin}.

Reviewer 3: An even better option (for another PR) would be to use the \SI{<value>}{<unit>} function, which would negate the need for many $$.

Reviewer 2: > there are some pretty odd uses of \si in the ASPECT manual right now

Yes, for sure.

Reviewer: Feel free to fix whatever you find!

Reviewer: These particular cases are now fixed in #3832.

> The surface temperature is set to $T_{\textit{surf}}$ = 1060 \si{\kelvin} and 

Reviewer: ```suggestion
The surface temperature is set to $T_{\text{surf}} = 1060 \si{\kelvin}$ and 
```

> the bottom temperature to $T_{\textit{cmb}}$ = 3450 \si{\kelvin}

Reviewer: ```suggestion
the bottom temperature to $T_{\text{cmb}} = 3450 \si{\kelvin}$
```

> The gravity vector is radial and its magnitude is $g$ = 10 \si{\metre\per\second\squared}.

Reviewer: ```suggestion
The gravity vector is radial and its magnitude is $g = 10 \si{\metre\per\second\squared}$.
```

Reviewer 3: Why do you favour putting the \si inside the $$? It works fine outside, and I prefer the kerning when it's not in the math environment.

Reviewer: I think they logically go together. I can compromise on where the units are, but I think the `=10` should be inside the formula environment :-)

Reviewer 3: Agreed on the `=10` ... that was really dumb of me.
If we put the \si{} into the math environment, we would need to manually separate number from unit with `\,` (otherwise we get 10m, rather than 10 m). So for now I've gone with the compromise option :)

Reviewer 3: @cedrict, I think I made all of the suggested changes to the main text and bib text in the overleaf doc, if you want to copy them over.

Reviewer: There is nothing ever "really dumb". We're all humans here.

Compromise sounds good to me.

> There is a single incompressible fluid in the domain, characterised by $\rho_0$ = 4500 \si{\kilogram\per\metre\cubed}, $\alpha$ = $2.5\cdot10^{-5}$ \si{\per\kelvin}, 

Reviewer: ```suggestion
There is a single incompressible fluid in the domain, characterized by $\rho_0$ = 4500 \si{\kilogram\per\metre\cubed}, $\alpha$ = $2.5\cdot10^{-5}$ \si{\per\kelvin}, 
```

> $k$ = 4 \si{\watt\per\metre\per\kelvin},

Reviewer: same here and everywhere below -- put the whole formula, including units, into the math environment. This makes sure the spacing around all math symbols is correct

$C_p$ = 1000 \si{\joule\per\kilogram\per\kelvin} and

> its internal heating rate is $Q_{\textit{int}}$ = $1\cdot10^{-12}$ \si{\watt\per\kilogram}.

Reviewer: Here and elsewhere:
```suggestion
its internal heating rate is $Q_{\text{int}}$ = $1\cdot10^{-12}$ \si{\watt\per\kilogram}.
```
The point is specifically that you want that text to *not* show up as italics, but rather as upright text because it's an abbreviation of natural language *text*, not formula variables (which should be shown as italics).

Reviewer 3: Huh, I always thought of the subscript as a part of the variable (because otherwise Q_int and Q_ext would be the same formula variable). But mostly I just dislike the kerning when the subscript is upright.

Reviewer 3: New discovery: \textsubscript and \textsuperscript can be used both inside and outside of the math environment (and they both use upright typeface).

Reviewer: The typographic community says that if you refer to matrix entries, use `A_{ij}`, which prints both `i` and `j` in italics because they are variable names, and puts a very small amount of space between the two of them to indicate that they are separate symbols. On the other hand, if you have a matrix `A_{stress}`, it should be written as `A_\text{stress}` to indicate that you don't have separate indices `s`, `t`, `r`, `e`, `s`, `s` that either serve as separate indices or a product of them, but rather that you are using a natural-language word. `\text` marks things up in this way: as *text*, rather than *variables in a formula*.

> The limit between the upper mantle (viscosity $\eta_{um}$)

Reviewer: ```suggestion
The interface between the upper mantle (viscosity $\eta_\text{um}$)
```

> and the lower mantle (viscosity $\eta_{lm}$) is fixed at $670\si{\km}$ depth.

Reviewer: ```suggestion
and the lower mantle (viscosity $\eta_\text{lm}$) is fixed at $670\si{\km}$ depth.
```

Reviewer 3: I don't think \si recognises \km. it just renders the text as-is, which in this case is ok, I suppose. I would prefer \kilo\meter.

As in the article we consider four time-independent radial viscosity profiles.

\begin{itemize}

> \item[a)] Isoviscous mantle, $\eta_{um}=\eta_{lm}=1.7\cdot 10^{24}$ \si{\pascal}

Reviewer: mark up `_{um}` and `_{lm}` as above, same for the next 3 lines

Reviewer: Also, shouldn't the units of viscosity be pascal * second?

\item[b)] $\eta_{um}=5.8\cdot 10^{22}$ \si{\pascal}, $\eta_{lm}=30\eta_{um}$ \si{\pascal}
\item[c)] Isoviscous mantle, $\eta_{um}=\eta_{lm}=5.8\cdot 10^{22}$ \si{\pascal}
\item[d)] $\eta_{um}=7\cdot 10^{21}$ \si{\pascal},  $\eta_{lm}=30\eta_{um}$ \si{\pascal}
\end{itemize}

> A separate ascii file {\tt visc\_depth\_X.txt} with {\tt X=\{a,b,c,d\}} contains each of these viscosity profiles. The resulting temperature field after 5 billion years of convection is shown in Fig.~\ref{fig:bunge_et_al}. Similar to the results obtained by \cite{burb96}, models in which the lower mantle is more viscous than the upper mantle are distinctly colder than their isoviscous equivalents, with more clearly defined upwellings. You can find a movie of how the temperature evolves over this time period at \url{https://youtu.be/5SPCU1sFGGc}.

Reviewer: ```suggestion
Separate ascii files {\tt visc\_depth\_X.txt} with {\tt X=\{a,b,c,d\}} contain each of these viscosity profiles. The resulting temperature fields after 5 billion years of convection are shown in Fig.~\ref{fig:bunge_et_al}. Similar to the results obtained by \cite{burb96}, models in which the lower mantle is more viscous than the upper mantle are distinctly colder than their isoviscous equivalents, with more clearly defined upwellings. You can find a movie of how the temperature evolves over this time period at \url{https://youtu.be/5SPCU1sFGGc}.
```

\begin{figure}[tbp]
  \centering
  \includegraphics[width=0.95\linewidth]{cookbooks/bunge_et_al_mantle_convection/temps.png}

>   \caption{\it From left to right: temperature field at time $t=5\cdot 10^9$ years obtained with viscosity profiles a, b, c and d.}

Reviewer: ```suggestion
  \caption{\it Bunge et al.~benchmark. From left to right: temperature field at time $t=5\cdot 10^9$ years obtained with viscosity profiles a, b, c and d.}
```
